src/main/java/com/faforever/api/data/domain/Name.java

```diff
@@ -0,0 +1,47 @@
+package com.faforever.api.data.domain;
+
+import com.yahoo.elide.annotation.DeletePermission;
+import com.yahoo.elide.annotation.Include;
+import com.yahoo.elide.annotation.UpdatePermission;
+import lombok.Setter;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.time.OffsetDateTime;
+
+@Entity
+@Table(name = "name_history")
+@Include(rootLevel = true, type = "name")
+@DeletePermission(expression = "Prefab.Role.None")
+@UpdatePermission(expression = "Prefab.Role.None")
+@Setter
+public class Name {
```

Comment:
`NameRecord` sounds good. We never use the term `Nickname` anywhere, so we shouldn't introduce that. **If** anything, then it would be `UsernameRecord`.